modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/ProgressInstantSpeedCalculation.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.pagemem;
+
+/**
+ * Represents instant speed calculation of some progress starting from zero.
+ */
+class ProgressInstantSpeedCalculation {
```

Comment:
All this class does is wrapping an `IntervalBasedMeasurement` in order to rename its methods. This may be an overkill

Reply:
I agree with @sashapolo, this class is a very thin wrapper around `IntervalBasedMeasurement`, using the measurement directly shouldn't hurt code readability that much.

Reply:
The thing is that `IntervalBasedMeasurement` is pretty non-obvious and involved. To understand what it does, you need to read a good part of it. In this concrete place in the code, only small fraction of its functionality is used: just computation of progress speed, where we provide the explicit progress value ourselves. All the averaging and ability to remember history is not used here, but it's not that obvious.

I introduced this adapter to clarify what it does. It has a clear (hopefully) javadoc both on the class and methods. It is just for computing progress speed, that's all. So, in my opinion, it does add value to the code: it hides complexity.

What do you think?

Reply:
your reasoning is good, adding more clarity in the existing code is cool. You can leave this class if you want to

Reply:
Ok, I'm leaving the class for now, and I tried to improve its (and its methods') names and added  more javadoc